File: backend/src/schemas/training_plan.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class TrainingPlanItemDetail(BaseModel):
    id: int
    problem_id: int
    title: str = ''
    company: str = ''
    difficulty: str = ''
    category_slug: str = ''
    tags: list[str] = Field(default_factory=list)
    sort_order: int = 1
    status: str = '未开始'


class TrainingPlanDetail(BaseModel):
    id: int
    name: str
    plan_type: str
    duration_days: int = 7
    total_problems: int = 0
    completed_count: int = 0
    correct_count: int = 0
    created_at: str
    updated_at: str
    items: list[TrainingPlanItemDetail] = Field(default_factory=list)


class TrainingPlanListItem(BaseModel):
    id: int
    name: str
    plan_type: str
    duration_days: int = 7
    total_problems: int = 0
    completed_count: int = 0
    correct_count: int = 0
    created_at: str
    updated_at: str


class TrainingPlanCreate(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)

    name: str = Field(min_length=1)
    plan_type: str = 'comprehensive'
    duration_days: int = Field(default=7, ge=1, le=90)
    problem_ids: list[int] = Field(min_length=1)


class PlanPreviewProblem(BaseModel):
    problem_id: int
    title: str = ''
    company: str = ''
    difficulty: str = ''
    category_slug: str = ''
    tags: list[str] = Field(default_factory=list)


class PlanPreview(BaseModel):
    name: str
    plan_type: str
    duration_days: int = 7
    problems: list[PlanPreviewProblem] = Field(default_factory=list)

File: frontend/src/types/trainingPlan.ts
export interface TrainingPlanItem {
  id: number
  problem_id: number
  title: string
  company: string
  difficulty: string
  category_slug: string
  tags: string[]
  sort_order: number
  status: '未开始' | '已通过' | '待复盘'
}

export interface TrainingPlanDetail {
  id: number
  name: string
  plan_type: string
  duration_days: number
  total_problems: number
  completed_count: number
  correct_count: number
  created_at: string
  updated_at: string
  items: TrainingPlanItem[]
}

export interface TrainingPlanListItem {
  id: number
  name: string
  plan_type: string
  duration_days: number
  total_problems: number
  completed_count: number
  correct_count: number
  created_at: string
  updated_at: string
}

export interface PlanPreviewProblem {
  problem_id: number
  title: string
  company: string
  difficulty: string
  category_slug: string
  tags: string[]
}

export interface PlanPreview {
  name: string
  plan_type: string
  duration_days: number
  problems: PlanPreviewProblem[]
}
